app_allocator/classes/reads_criterion.py

```diff
@@ -10,3 +11,6 @@ def __init__(self, name):
 
     def as_need(self, application):
         return ReadNeed(count=self.count)
+
+    def add_option(self, option, count, weight):
+        self.option_specs.add(OptionSpec(option, count, weight))
```

Comment:
Not a blocker, but Demeter would be happier with self.add_option.  I was kind of expecting that given that self.option_specs turned into a more complex thing (SortedList).  It seems a bit odd that the Criterion object declares option_specs and then expects other classes to know how to use it.

Reply:
Hm. I think we could move this up to Criterion, but that already has self.add_option defined to do something that I'm pretty sure I don't understand. (it doesn't add an option...)
If that's not a thing we want to keep, then this seems like reasonable default behavior for the base class. Is that what you were looking for?